### PR TITLE
Adapt script to also work on macOS

### PR DESCRIPTION
**Pull request by rfagen-zoox**
New laptop, who dis?

### CODE REVIEW

File: ssh-agent-canonicalize [hunk 1]
```diff
@@ -1,10 +1,19 @@
 #!/usr/bin/env bash
 set -euo pipefail
 
+# Get userid on MacOS
+if [[ $OSTYPE == 'darwin'* ]]; then
```
Comment (chpatton013) [inline ssh-agent-canonicalize]: The BSD options should work fine for the GNU versions of these commands. Let's just simplify it to use the short-opts. Then we don't need any variables or anything; we can just update the use of the args below.

File: ssh-agent-canonicalize [hunk 2]
```diff
@@ -15,22 +24,22 @@ function canon_agent_pid() {
...
-    if [ -S "$SSH_AUTH_SOCK" ]; then
+    if [[ -S "${SSH_AUTH_SOCK:-}" ]]; then
       echo ${SSH_AGENT_PID:-}
     fi
   )
 }
 
 function non_canon_agent_pids() {
-  echo $(pgrep ssh-agent) | sed -e "s#\b$(canon_agent_pid)\b##"
+  echo $( pgrep ssh-agent | grep -v "^$(canon_agent_pid)$" )
```
Comment (chpatton013) [inline ssh-agent-canonicalize]: I don't remember the history of why I made this function so complicated, but we can actually go one step further and remove the echo:
```
pgrep ssh-agent | grep -Ev "^$(canon_agent_pid)$"
```

**Review by chpatton013 — CHANGES_REQUESTED** [review]

**Comment by rfagen-zoox** [comment]
Ok. Finally found a moment. Cleaned up, tested on both 14.04 (lol) and macOS bash/zsh

**Comment by rfagen-zoox** [comment]
Otherwise, I believe the function `non_canon_agent_pids` won't actually return anything and causes the script to bail due to the `set -euo pipefail` ?

**Review by chpatton013 — APPROVED** [review]

**Pull request merged**